lib/util/lifted/influx/httpd/handler_metrics.go

```diff
@@ -0,0 +1,263 @@
+	err := registry.Register(openGeminiCollector)
+	if err != nil {
+		log.Fatal("Failed to register collector:", err)
+	}
+
+	metricsHandler = promhttp.HandlerFor(registry, promhttp.HandlerOpts{Registry: registry})
+}
+
+type OpenGeminiCollector struct {
+	*metrics.BaseCollector
+	indexMap map[string]*metrics.ModuleIndex
+}
+
+func NewOpenGeminiCollector() *OpenGeminiCollector {
+	c := &OpenGeminiCollector{
+		BaseCollector: metrics.NewBaseCollector(metrics.ComIndexRegistry),
+		indexMap:      make(map[string]*metrics.ModuleIndex),
+	}
+	// Initialize metrics
+	for moduleName, index := range c.IndexRegistry {
+		desc := make(map[string]*prometheus.Desc)
+		for indexName, help := range index.HelpMap {
+			desc[fmt.Sprintf("last_%s", indexName)] = metrics.NewDesc("", indexName, help, index.Labels)
+		}
+		c.AllModulesDesc[moduleName] = desc
+	}
+
+	return c
+}
+
+func (c *OpenGeminiCollector) Collect(ch chan<- prometheus.Metric) {
+	var indexValue interface{}
+	indexMap := c.indexMap
+	for moduleName, index := range c.IndexRegistry {
+		value, ok := indexMap[moduleName]
+		if ok {
+			// Get the label of the module
+			labelValues := make([]string, 0)
+			for _, labelName := range index.Labels {
+				labelValue, ok := value.LabelValues[labelName]
+				if !ok {
+					labelValues = append(labelValues, "nil")
+				}
+
+				labelValues = append(labelValues, labelValue)
+			}
+
+			for indexName := range index.HelpMap {
+				indexName = fmt.Sprintf("last_%s", indexName)
+				indexValue, ok = value.MetricsMap[indexName]
+				// If the indicator exists, write the indicator
+				if ok {
+					ch <- prometheus.MustNewConstMetric(c.AllModulesDesc[moduleName][indexName], prometheus.GaugeValue,
+						indexValue.(float64), labelValues...)
+				}
+			}
+		}
+	}
+
+}
+
+func (h *Handler) serveMetrics(w http.ResponseWriter, r *http.Request, user meta2.User) {
+	for moduleName, index := range openGeminiCollector.IndexRegistry {
+		moduleIndex, err := getMetrics(h, r, user, moduleName, index.Labels)
+		if err != nil {
+			continue
+		}
+		openGeminiCollector.indexMap[moduleName] = moduleIndex
+	}
+
+	metricsHandler.ServeHTTP(w, r)
+}
+
+// serverProm
+func getMetrics(h *Handler, r *http.Request, user meta2.User, tableName string, labelNames []string) (*metrics.ModuleIndex, error) {
+	// Retrieve the underlying ResponseWriter or initialize our own.
+	start := time.Now()
+
+	nodeID, _ := strconv.ParseUint(r.FormValue("node_id"), 10, 64)
+	var q *influxql.Query
+	var err error
+	sql := fmt.Sprintf("select last(*) from %s", tableName)
+	for i, labelName := range labelNames {
+		if i == 0 {
+			sql = fmt.Sprintf("%s group by %s", sql, labelName)
+		} else {
+			sql = fmt.Sprintf("%s,%s", sql, labelName)
+		}
+	}
+
+	qr := strings.NewReader(sql)
+	q, err, _ = h.getSqlQuery(r, qr)
+	if err != nil {
+		return nil, err
+	}
+
+	db := h.SQLConfig.Monitor.StoreDatabase
+	var qDuration *statistics.SQLSlowQueryStatistics
+	if !isInternalDatabase(db) {
```

Comment:
```suggestion
		if !isInternalDatabase(db) {
		qDuration = statistics.NewSqlSlowQueryStatistics(db)
		defer func() {
			d := time.Now().Sub(start)
			if d.Nanoseconds() > time.Second.Nanoseconds()*10 {
				qDuration.AddDuration("TotalDuration", d.Nanoseconds())
				statistics.AppendSqlQueryDuration(qDuration)
				h.Logger.Info("slow query", zap.Duration("duration", d), zap.String("db", qDuration.DB),
					zap.String("query", qDuration.Query))
			}
		}()
	}
```
The db name must be "_internal", this code can be deleted. The code record slowly sql for business db.

Reply:
I have removed.